docs/design/2022-06-15-phyical plan explain.md

```diff
@@ -0,0 +1,406 @@
+```
+
+```SQL
+SELECT * FROM t2 where a>9223372036854775807 and b>'aa'
+```
+
+**Before:**
+
+
+
+**After:**
+
+```
+== Physical Plan ==
+*(1) ColumnarToRow
++- TiSpark RegionTaskExec{downgradeThreshold=1000000000,downgradeFilter=[[b@VARCHAR(255) GREATER_THAN "aa"], [a@UNSIGNED LONG GREATER_THAN 9223372036854775807]]
+   +- RowToColumnar
+      +- TiKV FetchHandleRDD{[table: t2] IndexScan, Columns: a@UNSIGNED LONG, b@VARCHAR(255), c@VARCHAR(255): { IndexRangeScan: [Index: primary (a,b)]: { KeyRange: { RangeFliter: {[a@UNSIGNED LONG GREATER_THAN 9223372036854775807]}, Range: [([t\200\000\000\000\000\000\000\213_i\200\000\000\000\000\000\000\001\004\200\000\000\000\000\000\000\000], [t\200\000\000\000\000\000\000\213_i\200\000\000\000\000\000\000\001\372])], Selection: [b@VARCHAR(255) GREATER_THAN "aa"] }, TableRowIDScan:{ Selection: [b@VARCHAR(255) GREATER_THAN "aa"] } }, startTs: 433926839385194497}
+```
+
+### CoveringIndexScan
+
+* Remove Residual Filter
+* Rename PushDown Filter to Selection.
+* Add the description information of  Index.
+* Add RangeFliter to indicate which conditions are used to build Range.If RangeFilter is empty, it means the scan range is all, otherwise the scan range consists of the conditions in RangeFilter.
+
+```SQL
+CREATE TABLE `t2` (
+  `a` BIGINT(20) UNSIGNED  NOT NULL,
+  `b` varchar(255) NOT NULL,
+  `c` varchar(255) DEFAULT NULL,
+  PRIMARY KEY (`a`,`b`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+```
+
+```SQL 
+SELECT a FROM t2 where a>9223372036854775807 and b>'aa'
+```
+
+**Before:**
+
+
+
+**After:**
+
+```
+== Physical Plan ==
+*(1) Project [a#163]
++- *(1) ColumnarToRow
+   +- TiKV CoprocessorRDD{[table: t2] CoveringIndexScan, Columns: a@UNSIGNED LONG, b@VARCHAR(255): { IndexRangeScan: [Index: primary (a,b)]: { KeyRange: { RangeFliter: {[a@UNSIGNED LONG GREATER_THAN 9223372036854775807], [a@UNSIGNED LONG GREATER_THAN 9223372036854775807]}, Range: [([t\200\000\000\000\000\000\000\224_i\200\000\000\000\000\000\000\001\004\200\000\000\000\000\000\000\000], [t\200\000\000\000\000\000\000\224_i\200\000\000\000\000\000\000\001\372])], Selection: [b@VARCHAR(255) GREATER_THAN "aa"] } }, startTs: 433926993158864898}
+```
+
+## Test Design
+
+### TableScan
+
+1. Full Table Scan with Selection
+
+   ```SQL
+   CREATE TABLE `t1` (
+     `a` BIGINT(20) UNSIGNED  NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   ```SQL
+   SELECT * FROM t1 where a>9223372036854775807
+   ```
+
+2. Range Table Scan with Selection
+
+   ```SQL
+   CREATE TABLE `t1` (
+     `a` BIGINT(20) UNSIGNED NOT NULL,
+     `b` varchar(255) NOT NULL,
+     `c` varchar(255) DEFAULT NULL,
+     PRIMARY KEY (`a`) clustered
+   ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+   ```
+
+   ```SQL
+   SELECT * FROM t1 where a>9223372036854775807 and b > 'aa'
+   ```
+   
+3. prefix index scan
```

Comment:
what is the prefix index scan?

Reply:
range scan in a prefix index.change description.PTAL\